.ci/azure-pipelines.yml

```diff
@@ -23,6 +23,11 @@ jobs:
     pool:
       vmImage: 'ubuntu-latest'
 
+    variables:
+      - name: JELLYFIN_VERSION
+        value: replace(variables['Build.SourceBranch'], 'refs/tags/', '')
+        condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/v')
```

Comment:
It's not possible to place a condition on a variable definition.

Reply:
@EraYaN do you have some pointer for me on how to set this variable only when the condition evaluates to true? I think we should use a script block for that (like it was before) but I'm not sure what the syntax would be for setting the variable with to the result of an Azure expression.

Reply:
May I ask why set the Variable in any job besides `Publish` anyway? (It does not seem to be used besides in the `Publish` step job)

Reply:
Just put the `replace(variables['Build.SourceBranch'], 'refs/tags/', '')` bit in the place it's used. That should be only executed when the condition is true.

Reply:
The app is slightly different from the library. In the case of the app we need the tag set when building to include the version in the AndroidManifest, this is not needed for the library.
The build task should only have the variable set when it's building for a tag, not when building for PR's/master.

At line 97 I didn't need the condition because it's defined at the job-level. I do need a way to only create the variable on this job though.

Reply:
I see, let me check the Azure pipelines docu maybe i find something useful here :wink:

Reply:
@nielsvanvelzen Is the variable already needed in the `Test` job or does the `Build` job?

Reply:
Only in the build job and publish jobs not for test (although if it's easier to define it top-level it won't hurt)

Reply:
You can use YAML if constructs to define it I guess

Reply:
(Use 1 level of extra indent after the if line)

Reply:
I now moved the declaration of the variable top-level with the if-statement so it is available in all jobs/steps. This way there is no duplicate code when we need it in multiple jobs.